tests/foreman/cli/test_subscription.py

```diff
@@ -189,3 +199,169 @@ def test_negative_manifest_refresh(self):
             Subscription.refresh_manifest({
                 'organization-id': self.org['id'],
             })
+
+    @tier3
+    def test_positive_restore_ak_and_content_hosts_subscriptions(self):
+        """Restore activation key and content hosts subscriptions
+
+        :id: a44fdeda-9c8c-4316-85b4-a9b6b9f1ffdb
+
+        :steps:
+            1. Setup activation key , lifecycle environment and content view
```

Comment:
Where did you get these steps? I read BZ information and it has some deviations from your steps

Reply:
Especially that csv was edited to another host/capsules before applying

Reply:
@oshtaier to make the exact BZ steps we have to be able to get an other manifest with the same subscriptions but with different agreement number, that is not realizable with the current infra implementation (we need muti-manifests s to be supported), the current  coverage implement that by deleting completely the manifest and re-import all data, if you insist on the exact implementation we have to make this PR in do not merge state (marked as blocked ) and wait that PR implementation (multi-manifest) to be merged.

Reply:
Hmm. Coverage that you provided in that PR is also needed, but still I am not sure that it is actually cover mentioned BZ

Reply:
+1 to @oshtaier . The steps are looking different to me especially without changing subscription id in csv.
We need @JacobCallahan approval whether these steps fully cover the BZ, and if not - corresponding framework expansion is needed.

Reply:
@abalakh @ldjebran There is one primary case that should be covered in the BZ, which the current test does not cover. The ideal steps for the bug would be:
1. Import a manifest, with a couple of subscriptions.
2. Register at least one host (preferably more).
3. Attach the same subscription to all registered hosts.
4. Export the content hosts to a csv file.
5. Modify the csv file to change the hosts from their current subscription to a different one.
6. Import the csv file
Result: The hosts should no longer have the old subscription, and now have the new one.

There is also an additional test case that could be written for what I did in comment 36. This is pretty similar to the steps above, but does not require hosts to be registered.
1. Upload a manifest with a couple of subscriptions.
2. Create an activation key, aligning one of the subscriptions to it.
3. Export the activation key to a csv file.
4. Modify the csv file to change the activation key's subscriptions from the previous sub to a different one.
5. Import the csv file.
Result: The activation key should not have the old subscription, and should have the new one.

With that said, this is still a very valid test that we should have in the code, just not aligned to this bug.